WordPress/gutenberg
# Move the directives and markup needed for the lightbox to block supports

_Marking this as a draft because the code will need to be updated with the changes made in [this other pull request](https://github.com/WordPress/gutenberg/pull/51206)._

## What?
Instead of having the directives and the markup needed to make the lightbox behavior work in the Image block `render_callback`, this pull request moves them to a `render_block` filter called in the block supports.

## Why?
There are mainly two reasons for this pull request:
* Right now, if there are changes applied to the Image markup through a `render_block` filter (like [the duotone](https://github.com/WordPress/gutenberg/issues/50973) is doing adding a class), those changes are not applied to the markup. It would solve [this bug](https://github.com/WordPress/gutenberg/issues/50973).
* It would be a first step of abstracting the lightbox behavior so it is not completely coupled to the Image block and can be easily reused.
 
Note that this pull request doesn't aim to completely abstract the lightbox behavior, it would be just a good first step. There are some things missing in that regard that I believe could be handled in a subsequent pull request:
* Move the styles and the scripts outside of the Image block.
* Modify the markup, directives, and styles to make it work properly for other blocks apart from the core/image.

## How?
* Remove the directives and markup from the `render_callback` of the image block.
* Register a new "block supports" for `supports.behaviors.lightbox` using `WP_Block_Supports::get_instance()->register`.
* When the behaviors support is registered, if the block has the lightbox support set to true, add a `render_block_{$this->name}` filter to add the directives and the markup.
* Update the editor hook to read the block supports.

## Testing Instructions
* Enable the "Interactivity API and Behaviors UI" in Gutenberg->Experiments.
* Check that the lightbox keeps working as expected.
* Add a duotone filter to some of your images and check that it works.

## Video explanation

https://www.loom.com/share/09ffb096070943d7b73ed0d885c40fd4
